smac/utils/validate.py

```diff
@@ -189,43 +203,137 @@ def _validate_parallel(self, tae, runs, n_jobs, backend):
+                                          run.inst_specs,
                                           capped=False) for run in runs)
         return run_results
 
-    def get_runs(self, configs, insts, repetitions=1, runhistory=None):
+    def validate_epm(self, config_mode:Union[str, typing.List[Configuration]]='def',
+                     instance_mode:Union[str, typing.List[str]]='test',
+                     repetitions:int=1, runhistory:RunHistory=None) -> RunHistory:
+        """
+        Use EPM to predict costs/runtimes for unknown config/inst-pairs.
+
+        Parameters
+        ----------
+        config_mode: str or list<Configuration>
+            what configurations to validate
+            either from [def, inc, def+inc, time, all], time means evaluation at
+            timesteps 2^-4, 2^-3, 2^-2, 2^-1, 2^0, 2^1, ...
+            or directly a list of Configuration
+        instance_mode: str or list<str>
+            what instances to use for validation, either from
+            [train, test, train+test] or directly a list of instances
+        repetitions: int
+            number of repetitions in nondeterministic algorithms
+        runhistory: RunHistory
+            optional, RunHistory-object to reuse runs
+
+        Returns
+        -------
+        runhistory: RunHistory
+            runhistory with predicted runs
+        """
+        if not isinstance(runhistory, RunHistory):
+            raise ValueError("No runhistory specified for validating with EPM!")
+        # Train random forest and transform training data (from given rh)
+        rh2epm = RunHistory2EPM4Cost(num_params=len(self.scen.cs.get_hyperparameters()),
+                                     scenario=self.scen, rng=self.rng)
+        X, y = rh2epm.transform(runhistory)
+        self.logger.debug("Training model with data of shape X: %s, y:%s",
+                          str(X.shape), str(y.shape))
+
+        types, bounds = get_types(self.scen.cs, self.scen.feature_array)
+        model = RandomForestWithInstances(types=types,
+                                          bounds=bounds,
+                                          instance_features=self.scen.feature_array,
+                                          seed=self.rng.randint(MAXINT),
+                                          ratio_features=1.0)
+        model.train(X, y)
+
+        # Predict desired runs
+        runs = self.get_runs(config_mode, instance_mode, repetitions, runhistory)
+        try:
+            feature_array_size = len(self.scen.cs.get_hyperparameters()) + self.scen.feature_array.shape[1]
+        except AttributeError:
+            feature_array_size = len(self.scen.cs.get_hyperparameters())
+        X_pred = np.empty((len(runs), feature_array_size))
+        for idx, run in enumerate(runs):
+            if hasattr(self.scen, "feature_dict") and run.inst != None:
+                X_pred[idx] = np.hstack([run.config.get_array(),
```

Comment:
I think you have to use `from smac.configspace import convert_configurations_to_array` such that the inactive parameters are also correctly imputed

Reply:
Thanks. Changed.